americangut/notebook_environment.py

```diff
@@ -0,0 +1,357 @@
+    'ag-gg-100nt-1k-unifrac-pc':
+        ('6/ag-pgp-hmp-gg-100nt-1k-bdiv/'
+         'unweighted_unifrac_ag-gg-100nt-1k-pc.txt'),
+
+    'ag-gg-100nt-1k-bdiv-subsampled-unifrac':
+        ('6/ag-pgp-hmp-gg-100nt-1k-bdiv/'
+         'unweighted_unifrac_ag-gg-100nt-1k-subsampled.txt'),
+    'ag-gg-100nt-1k-subsampled-unifrac-pc':
+        ('6/ag-pgp-hmp-gg-100nt-1k-bdiv/'
+         'unweighted_unifrac_ag-gg-100nt-1k-subsampled-pc.txt'),
+
+    'ag-gg-100nt-1k-bdiv-wunifrac':
+        ('6/ag-pgp-hmp-gg-100nt-1k-bdiv/'
+         'weighted_unifrac_ag-gg-100nt-1k.txt'),
+    'ag-gg-100nt-1k-wunifrac-pc':
+        ('6/ag-pgp-hmp-gg-100nt-1k-bdiv/'
+         'weighted_unifrac_ag-gg-100nt-1k-pc.txt'),
+
+    # taxonomy summaries
+    'ag-taxa': '7/taxa',
+    'ag-L2-taxa-tsv': '7/taxa/otu_table_L2.txt',
+    'ag-L2-taxa-biom': '7/taxa/otu_table_L2.biom',
+    'ag-L2-taxa-md': '7/taxa/ag-cleaned_L2.txt',
+    'ag-L3-taxa-tsv': '7/taxa/otu_table_L3.txt',
+    'ag-L3-taxa-biom': '7/taxa/otu_table_L3.biom',
+    'ag-L6-taxa-tsv': '7/taxa/otu_table_L6.txt',
+    'ag-L6-taxa-biom': '7/taxa/otu_table_L6.biom',
+    'ag-L2-taxa-skin-biom': '7/taxa/otu_table_skin_L2.biom',
+    'ag-L2-taxa-oral-biom': '7/taxa/otu_table_oral_L2.biom',
+    'ag-L2-taxa-fecal-biom': '7/taxa/otu_table_fecal_L2.biom',
+    'ag-L6-taxa-skin-biom': '7/taxa/otu_table_skin_L6.biom',
+    'ag-L6-taxa-oral-biom': '7/taxa/otu_table_oral_L6.biom',
+    'ag-L6-taxa-fecal-biom': '7/taxa/otu_table_fecal_L6.biom',
+
+    # collapsed samples
+    'ag-100nt-1k-biom': '7.5/ag-100nt-1k.biom',
+    'ag-100nt-1k-fecal-biom': '7.5/ag-100nt-1k-fecal.biom',
+    'ag-100nt-1k-skin-biom': '7.5/ag-100nt-1k-oral.biom',
+    'ag-100nt-1k-oral-biom': '7.5/ag-100nt-1k-skin.biom',
+
+    'ag-100nt-1k-fecal-sex-biom':  '7.5/ag-100nt-1k-fecal-sex.biom',
+    'ag-100nt-1k-fecal-diet-biom': '7.5/ag-100nt-1k-fecal-diet.biom',
+    'ag-100nt-1k-fecal-age-biom':  '7.5/ag-100nt-1k-fecal-age.biom',
+    'ag-100nt-1k-fecal-bmi-biom':  '7.5/ag-100nt-1k-fecal-bmi.biom',
+
+    'ag-100nt-1k-oral-sex-biom':  '7.5/ag-100nt-1k-oral-sex.biom',
+    'ag-100nt-1k-oral-diet-biom': '7.5/ag-100nt-1k-oral-diet.biom',
+    'ag-100nt-1k-oral-age-biom':  '7.5/ag-100nt-1k-oral-age.biom',
+    'ag-100nt-1k-oral-flossing-biom':  '7.5/ag-100nt-1k-oral-flossing.biom',
+
+    'ag-100nt-1k-skin-sex-biom':  '7.5/ag-100nt-1k-skin-sex.biom',
+    'ag-100nt-1k-skin-cosmetics-biom': '7.5/ag-100nt-1k-skin-cosmetics.biom',
+    'ag-100nt-1k-skin-age-biom':  '7.5/ag-100nt-1k-skin-age.biom',
+    'ag-100nt-1k-skin-hand-biom':  '7.5/ag-100nt-1k-skin-hand.biom',
+
+    # per-sample results
+    'successful-ids': '8/successful_ids.txt',
+    'unsuccessful-ids': '8/unsuccessful_ids.txt',
+    'per-sample-results': '8/per-sample-results',
+}
+
+
+def _assert_environment():
+    if qiime.__version__ != '1.9.1':
+        obs_version = qiime.__version__
+        raise ImportError(("QIIME 1.9.1 is not in the environment, found "
```

Comment:
Since system calls are made to QIIME, and are made somewhat late in
execution, the intent is to fail early and immediately if the environment
does not appear to be sane.

On Wed, Oct 21, 2015 at 12:00 PM, Antonio Gonzalez <notifications@github.com
> wrote:

> In americangut/notebook_environment.py
> <https://github.com/biocore/American-Gut/pull/162#discussion_r42659106>:
>
> > +    'ag-100nt-1k-skin-sex-biom':  '7.5/ag-100nt-1k-skin-sex.biom',
> > +    'ag-100nt-1k-skin-cosmetics-biom': '7.5/ag-100nt-1k-skin-cosmetics.biom',
> > +    'ag-100nt-1k-skin-age-biom':  '7.5/ag-100nt-1k-skin-age.biom',
> > +    'ag-100nt-1k-skin-hand-biom':  '7.5/ag-100nt-1k-skin-hand.biom',
> > +
> > +    # per-sample results
> > +    'successful-ids': '8/successful_ids.txt',
> > +    'unsuccessful-ids': '8/unsuccessful_ids.txt',
> > +    'per-sample-results': '8/per-sample-results',
> > +}
> > +
> > +
> > +def _assert_environment():
> > +    if qiime.__version__ != '1.9.1':
> > +        obs_version = qiime.__version__
> > +        raise ImportError(("QIIME 1.9.1 is not in the environment, found "
>
> Just to be sure that are there? [image: :smile:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/American-Gut/pull/162/files#r42659106>.
>